### PR TITLE
Added the Team resource list_resources() method and a test

### PR DESCRIPTION
**Pull request by crleblanc**
# Citrine Python PR

## Description 
As per PLA-9753 this PR implements the listing of resources for a team via the `list_resoures` method on the Team class.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/team.py [hunk 1]
```diff
@@ -13,7 +13,9 @@
 WRITE = "WRITE"
 READ = "READ"
 SHARE = "SHARE"
-TEAM_ACTIONS = Union[WRITE, READ, SHARE]
+CREATE = "CREATE"
+PUBLISH = "PUBLISH"
+TEAM_ACTIONS = Union[WRITE, READ, SHARE, CREATE, PUBLISH]
```
Comment (crleblanc) [inline src/citrine/resources/team.py:18]: @zoesyang Can you confirm that these team actions should be added?  I've seen them mentioned elsewhere so I'm guessing they're missing from this file but would like to double check.

File: src/citrine/resources/team.py [hunk 2]
```diff
@@ -224,6 +226,30 @@ def un_share(self, *, resource: Resource, target_team_id: Union[str, UUID]) -> b
         )
         return True
 
+    def list_resources(self, resource_type: str, action: str) -> List[str]:
```
Comment (crleblanc) [inline src/citrine/resources/team.py:229]: I plan to replace 'str' with a better type here for resource_type and action.

**Review by kroenlein — COMMENTED** [review]
Non-blocking question.  Also, you've prepped an integration / nextgen-devkit e2e for this?

File: src/citrine/__version__.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = '1.33.1'
+__version__ = '1.34.1'
```
Comment (kroenlein) [inline src/citrine/__version__.py:1]: Interesting version bump...

Comment (crleblanc) [reply src/citrine/__version__.py:1]: I'm left guessing what you mean.  As per semver.org:

Given a version number MAJOR.MINOR.PATCH, increment the:

    MAJOR version when you make incompatible API changes,
    MINOR version when you add functionality in a backwards compatible manner, and
    PATCH version when you make backwards compatible bug fixes.

It's new functionality that is backwards compatible, hence a minor version bump.

File: src/citrine/resources/team.py [hunk 4]
```diff
@@ -224,6 +232,30 @@ def un_share(self, *, resource: Resource, target_team_id: Union[str, UUID]) -> b
...
+        -------
+        List[str]
+            The ids of the resources in this team where the owner has the specified action
+
+        """
+        query_params = {"domain": self._path(), "action": action}
+        return self.session.get_resource(f"/{resource_type}/authorized-ids",
+                                         params=query_params,
+                                         version="v3")['ids']
+
```
[the remark this answers is not in the document]
Comment (crleblanc) [reply src/citrine/resources/team.py:258]: I see there was discussion about this approach on slack, but I didn't see the responses in the thread (ps: damn you Slack).  The API is `/api/v3/.teamId/authorized-ids` - hence the 1:1 mapping with teams.  I'm fine to close this PR and implement it the way you suggest, if this is the agreed approach.

**Pull request closed without merging**